action/action-dsl/src/main/java/com/powsybl/action/dsl/ast/ActionExpressionPrinter.java

```diff
@@ -21,20 +21,28 @@
 public class ActionExpressionPrinter extends ExpressionPrinter implements ActionExpressionVisitor<Void, Void> {
 
     public static String toString(ExpressionNode node) {
-        StringWriter writer = new StringWriter();
-        try (PrintStream os = new PrintStream(new WriterOutputStream(writer, StandardCharsets.UTF_8))) {
-            print(node, os);
+        try (StringWriter writer = new StringWriter()) {
+            write(node, writer);
+            return writer.toString();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
         }
+    }
 
-        return writer.toString();
+    private static void write(ExpressionNode node, StringWriter out) {
+        node.accept(new ActionExpressionPrinter(out), null);
     }
 
     public static void print(ExpressionNode node) {
         print(node, System.out);
     }
 
     public static void print(ExpressionNode node, PrintStream out) {
-        node.accept(new ActionExpressionPrinter(out), null);
+        out.print(toString(node));
+    }
+
+    public ActionExpressionPrinter(StringWriter out) {
+        super(out);
     }
```

Comment:
Same question here: do we really need a `StringWriter`? It seems a Writer is sufficient here. You should change constructors of both classes to take the highest interface you can from the inheritance tree.

Reply:
Agreed